File: backend/models/quote.go
package models

import (
	"github.com/shopspring/decimal"

	"time"

	"github.com/google/uuid"
)

type Quote struct {
	ID                 uuid.UUID       `gorm:"type:uuid;primaryKey" json:"id"`
	CompanyID          uuid.UUID       `gorm:"type:uuid;not null;index" json:"company_id"`
	CariID             uuid.UUID       `gorm:"type:uuid;not null;index" json:"cari_id"`
	Number             string          `gorm:"type:varchar(100);not null" json:"number"` // unique per company sequence-generated
	Date               time.Time       `gorm:"not null" json:"date"`
	ExpiryDate         time.Time       `gorm:"not null" json:"expiry_date"`
	Currency           string          `gorm:"type:varchar(10);not null;default:'TRY'" json:"currency"`
	ExchangeRate       decimal.Decimal `gorm:"type:numeric(65,6);not null;default:1" json:"exchange_rate"`
	Subtotal           decimal.Decimal `gorm:"type:numeric(65,4);not null;default:0" json:"subtotal"`
	DiscountTotal      decimal.Decimal `gorm:"type:numeric(65,4);not null;default:0" json:"discount_total"`
	TaxTotal           decimal.Decimal `gorm:"type:numeric(65,4);not null;default:0" json:"tax_total"`
	Total              decimal.Decimal `gorm:"type:numeric(65,4);not null;default:0" json:"total"`
	Status             string          `gorm:"type:varchar(50);not null;default:'draft'" json:"status"` // 'draft', 'sent', 'accepted', 'rejected', 'expired', 'converted'
	PublicToken        string          `gorm:"type:varchar(64);uniqueIndex" json:"-"`
	SentAt             *time.Time      `json:"sent_at"`
	LastSentAt         *time.Time      `json:"last_sent_at"`
	RejectNote         string          `gorm:"type:text" json:"reject_note"`
	RespondedAt        *time.Time      `json:"responded_at"`
	Note               string          `gorm:"type:text" json:"note"`
	ConvertedInvoiceID *uuid.UUID      `gorm:"type:uuid;index" json:"converted_invoice_id,omitempty"`
	Items              []QuoteItem     `gorm:"foreignKey:QuoteID;constraint:OnDelete:CASCADE" json:"items"`
	CreatedBy          *uuid.UUID      `gorm:"type:uuid" json:"created_by"`
	UpdatedBy          *uuid.UUID      `gorm:"type:uuid" json:"updated_by"`
	CreatedByUser      *User           `gorm:"foreignKey:CreatedBy;constraint:OnDelete:SET NULL" json:"created_by_user,omitempty"`
	UpdatedByUser      *User           `gorm:"foreignKey:UpdatedBy;constraint:OnDelete:SET NULL" json:"updated_by_user,omitempty"`
	CreatedAt          time.Time       `json:"created_at"`
	UpdatedAt          time.Time       `json:"updated_at"`
}

type QuoteItem struct {
	ID           uuid.UUID       `gorm:"type:uuid;primaryKey" json:"id"`
	CompanyID    uuid.UUID       `gorm:"type:uuid;not null;index" json:"company_id"`
	QuoteID      uuid.UUID       `gorm:"type:uuid;not null;index" json:"quote_id"`
	ProductID    *uuid.UUID      `gorm:"type:uuid" json:"product_id"`
	Description  string          `gorm:"type:text" json:"description"`
	Quantity     decimal.Decimal `gorm:"type:numeric(65,4);not null;default:1" json:"quantity"`
	Unit         string          `gorm:"type:varchar(50)" json:"unit"`
	UnitPrice    decimal.Decimal `gorm:"type:numeric(65,4);not null;default:0" json:"unit_price"`
	DiscountRate decimal.Decimal `gorm:"type:numeric(5,2);not null;default:0" json:"discount_rate"`
	TaxRate      decimal.Decimal `gorm:"type:numeric(5,2);not null;default:0" json:"tax_rate"`
	LineTotal    decimal.Decimal `gorm:"type:numeric(65,4);not null;default:0" json:"line_total"`
	Currency       string          `gorm:"type:varchar(10);not null;default:'TRY'" json:"currency"`
	ExchangeRate   decimal.Decimal `gorm:"type:numeric(65,10);not null;default:1" json:"exchange_rate"`
	ExchangeRateOp string          `gorm:"type:varchar(5);not null;default:'*'" json:"exchange_rate_op"`
	CreatedAt    time.Time       `json:"created_at"`
	UpdatedAt    time.Time       `json:"updated_at"`
}

File: frontend/src/stores/quote.ts
import { defineStore } from 'pinia'
import {
  getQuotesApi,
  getQuoteByIDApi,
  createQuoteApi,
  updateQuoteApi,
  deleteQuoteApi,
  updateQuoteStatusApi,
  convertQuoteApi,
  sendQuoteApi,
  bulkSendQuoteApi,
  getPublicQuoteApi,
  acceptPublicQuoteApi,
  rejectPublicQuoteApi,
} from '@/api/quote'

export interface QuoteItem {
  id: string
  company_id: string
  quote_id: string
  product_id: string | null
  description: string
  quantity: string
  unit: string
  unit_price: string
  discount_rate: string
  tax_rate: string
  line_total: string
  currency: string
  exchange_rate: string
  exchange_rate_op: string
}

import type { UserRef } from './invoice'

export interface Quote {
  id: string
  company_id: string
  cari_id: string
  type?: 'sales' | 'purchase'
  number: string
  date: string
  expiry_date: string
  currency: string
  exchange_rate: string
  subtotal: string
  discount_total: string
  tax_total: string
  total: string
  status: 'draft' | 'sent' | 'accepted' | 'rejected' | 'expired' | 'converted'
  note: string
  reject_note?: string
  responded_at?: string
  sent_at?: string
  last_sent_at?: string
  public_token?: string
  converted_invoice_id?: string
  items?: QuoteItem[]
  created_at: string
  updated_at: string
  created_by_user?: UserRef | null
  updated_by_user?: UserRef | null
}

export const useQuoteStore = defineStore('quote', {
  state: () => ({
    quotes: [] as Quote[],
    total: 0,
    activeQuote: null as Quote | null,
    loading: false,
  }),
  actions: {
    async fetchQuotes(params?: any) {
      this.loading = true
      try {
        const res = await getQuotesApi(params)
        this.quotes = res.data.data
        this.total = res.data.meta.total
      } finally {
        this.loading = false
      }
    },
    async fetchQuoteByID(id: string) {
      this.loading = true
      try {
        const res = await getQuoteByIDApi(id)
        this.activeQuote = res.data.data
      } finally {
        this.loading = false
      }
    },
    async createQuote(data: any) {
      const res = await createQuoteApi(data)
      return res.data.data
    },
    async updateQuote(id: string, data: any) {
      const res = await updateQuoteApi(id, data)
      return res.data.data
    },
    async deleteQuote(id: string) {
      await deleteQuoteApi(id)
    },
    async updateQuoteStatus(id: string, status: string) {
      await updateQuoteStatusApi(id, status)
    },
    async convertQuote(id: string) {
      const res = await convertQuoteApi(id)
      return res.data.data
    },
    async sendQuote(id: string) {
      const res = await sendQuoteApi(id)
      return res.data
    },
    async bulkSendQuote(ids: string[]) {
      const res = await bulkSendQuoteApi(ids)
      return res.data.data as { sent: string[]; failed: { id: string; error: string }[] }
    },
    async getPublicQuote(token: string) {
      const res = await getPublicQuoteApi(token)
      return res.data.data
    },
    async acceptPublicQuote(token: string) {
      const res = await acceptPublicQuoteApi(token)
      return res.data.data
    },
    async rejectPublicQuote(token: string, note: string) {
      const res = await rejectPublicQuoteApi(token, note)
      return res.data.data
    },
  },
})
